src/pull.go

```diff
@@ -51,8 +52,47 @@ func (g *Commands) Pull(byId bool) error {
 	cl, clashes, err := pullLikeResolve(g, byId)
 
 	if len(clashes) >= 1 {
-		warnClashesPersist(g.log, clashes)
-		return ErrClashesDetected
+		if g.opts.FixClashes {
+			g.log.Logln("Some clashes found")
+			if g.opts.canPrompt() {
+				if !promptForChanges("Try fixing them [Y/N] ? ") {
+					return ErrClashesDetected
+				}
+			}
+			sort.Sort(ByPath(clashes))
+
+			lastPath := ""
```

Comment:
I see what you are trying to do here, good one. Sorting first, getting paths close together then detecting when the last path has changed. However, we could actually cluster clashing files by their path and not have to manually keep track of these changes. Please take a look at this:

```go
func autoRenameClashes(g *Commands, clashes []*Change) error {
	clashesMap := map[string][]*Change{}

	scheduledRenamings := []keyValue{}

	for _, clash := range clashes {
		group, _ := clashesMap[clash.Path]
		group = append(group, clash)
		clashesMap[clash.Path] = group
	}

	for commonPath, group := range clashesMap {
		ext := filepath.Ext(commonPath)
		name := strings.TrimSuffix(commonPath, ext)

		for i, ch := range group {
			newPath := fmt.Sprintf("%v_%d%v", name, i, ext)
			scheduledRenamings = append(scheduledRenamings, keyValue{key: newPath, value: ch})
		}
	}

	successfulRenames := 0
	for _, kv := range scheduledRenamings {
		ch, ok := kv.value.(*Change)
		if !ok {
			continue
		}
		if ch.Src == nil {
			continue
		}

		newPath := kv.key
		dupCheck, err := g.rem.FindByPath(newPath)
		if err != nil && err != ErrPathNotExists {
			return err
		}
		if dupCheck != nil {
			continue
		}

		if rErr := g.rem.Rename(ch.Src.Id, newPath); rErr != nil {
			g.log.LogErrf("err: rename %v (%v) to %v failed with %v", ch.Path, ch.Src.Id, newPath, rErr)
		} else {
			successfulRenames += 1
		}
	}

	// do something here with successful renames and then terminate as you had previously
	err := ErrClashesDetected
	if successfulRenames >= 1 {
		err = fmt.Errorf("clashes were renamed, please try pulling again")
	}
	return err
}
```
Then use renameClashes
I haven't tested it but the idea is you can ensure correctness of clustered without having to do a sort paths